index.js

```diff
@@ -1245,6 +1250,13 @@ Client.prototype.supportsActivationMethodField = function () {
     return semver.gte(this._apmServerVersion, '8.7.1')
   }
 }
+Client.prototype.supportsConfiguredAndDetectedHostname = function () {
+  if (!this._apmServerVersion) {
+    return true // Optimistically assume APM server is >=7.4.
```

Comment:
Reviewer note: I'm considering changing this, but I'm not sure. It would be *nice* to just be able to assume we are on newer APM server... but I suppose for someone using APM server <7.4 *and* when fetching the APM server version fails (for whatever reason), we will no longer be sending `metadata.system.hostname`. I don't think that'll *break* anything, but it would be a change in the data. Thoughts?

Reply:
Not a GO developer but looking at [the changes in apm-server](https://github.com/elastic/apm-server/pull/2540/files#diff-db6b9d7c7ae6d368324efedf21092eca303205602ee5b259e737837b276f0cd1L27) if we sent `metadata.system.hostname` the server is going to drop it. So by returning `true` we are implicitly encouraging to update apm-server which IMO is good.

Reply:
Meh, I think it is fine as is.